azure-active-directory-server/src/main/java/org/jetbrains/teamcity/aad/AADAuthenticationScheme.java

```diff
@@ -121,7 +122,12 @@ public HttpAuthenticationResult processAuthenticationRequest(@NotNull HttpServle
     final String email = token.getClaim(EMAIL_CLAIM);
 
     final ServerPrincipal principal = myPrincipalFactory.getServerPrincipal(name, oid, email, schemeProperties);
-
+   
+    if(principal == null)
```

Comment:
hi @ekoshkin, thanks for your feedback. The user creation behavior worked fined without that check,  but I added that in order to display something relevant to the user instead of just reload the login form without any indication of what failed. Sometimes the user could authenticate with an email address but the _upn_ claim, that we're using here as the email to match an existent TC user, could point to an alias.